src/zmq/zmqnotificationinterface.cpp

```diff
@@ -87,7 +87,7 @@ bool CZMQNotificationInterface::Initialize()
         return false;
     }
 
-    std::list<CZMQAbstractNotifier*>::iterator it=notifiers.begin();
+    auto it=notifiers.begin();
```

Comment:
Why would you separate this into multiple lines? Do you understand what is going on in the `for` loop? If not, I would advise you to first learn `for` loop structure. This should be one line. Please also space around assignment and comparison operators. This is basic clean syntax. We need to maintain a standard of clean syntax in the codebase. We have to take code quality seriously, and these "small" nitpicky items actually do matter. 

`for (auto it = notifiers.begin(); it != notifiers.end(); ++it)`

Looking at this in a larger context, I'm not sure what you're trying to do in this loop. This might not be a good candidate for a range-based `for`, but I'll have to look closer to know what you're trying to do.

Reply:
Ahh, I see why the original person did it this way. There is a check on line 104 which checks everything completed, and used the iterator. Therefor it needs to be above the loop to be in scope. Whether that check is needed or not, I do not know, but that's why the var is declared there.

Reply:
Ah, ok, got it. I will have to review the entirety of changes instead of just per-commit, would probably help me grok this changeset more. Traveling today, but will do a fuller review once I'm able to look more closely at it.

Reply:
sounds good.